File: client/src/app/shared/models/order.ts
export type Order = {
  id: number;
  orderDate: Date;
  buyerEmail: string;
  shippingAddress: ShippingAddress;
  deliveryMethod: string;
  shippingPrice: number;
  paymentSummary: PaymentSummary;
  orderItems: OrderItem[];
  subtotal: number;
  status: string;
  total: number;
  paymentIntentId: string;
};

export type ShippingAddress = {
  name: string;
  line1: string;
  line2?: string;
  city: string;
  state: string;
  postalCode: string;
  country: string;
};

export type PaymentSummary = {
  last4: number;
  brand: string;
  expMonth: number;
  expYear: number;
};

export type OrderItem = {
  productId: number;
  name: string;
  description: string;
  pictureUrl?: string;
  brand: string;
  category: string;
  price: number;
  quantity: number;
};

export type OrderToCreate = {
  cartId: string;
  deliveryMethodId: number;
  shippingAddress: ShippingAddress;
  paymentSummary: PaymentSummary;
};

File: Skinet.Entities/OrderAggregate/Order.cs
using Skinet.Entities.Delivery;

namespace Skinet.Entities.OrderAggregate;

public class Order : BaseEntity
{
    public DateTime OrderDate { get; set; } = DateTime.UtcNow;

    public required string BuyerEmail { get; set; }

    public ShippingAddress ShippingAddress { get; set; } = null!;

    public DeliveryMethod DeliveryMethod { get; set; } = null!;

    public PaymentSummary PaymentSummary { get; set; } = null!;

    public decimal Discount { get; set; }
    public string? CouponCode { get; set; }

    public List<OrderItem> OrderItems { get; set; } = [];

    public decimal SubTotal { get; set; }

    public OrderStatus Status { get; set; } = OrderStatus.Pending;

    public required string PaymentIntentId { get; set; }

    public decimal GetTotal()
    {
        return SubTotal - Discount + DeliveryMethod.Price;
    }
}
